video/out/vo_gpu_next.c

```diff
@@ -1500,89 +1499,113 @@ static char *get_cache_file(struct priv *p, char *type)
     } else {
         dir = mp_find_user_file(NULL, p->global, "cache", "");
     }
-    if (dir && dir[0]) {
-        file = mp_path_join(NULL, dir, "libplacebo.cache");
+    if (dir && dir[0])
         mp_mkdirp(dir);
-    }
-    talloc_free(dir);
 done:
-    return file;
+    return dir;
 }
 
-static void load_cache_files(struct priv *p)
+static char *cache_filepath(void *ta_ctx, char *dir, uint64_t key)
 {
-    char *icc_cache = get_cache_file(p, "icc");
-    char *shader_cache = get_cache_file(p, "shader");
-    bool same_cache = false;
-    if (icc_cache && shader_cache)
-        same_cache = strcmp(icc_cache, shader_cache) == 0;
-    if (shader_cache) {
-        FILE *cache = fopen(shader_cache, "rb");
-        if (cache) {
-            int ret = pl_cache_load_file(p->shader_cache, cache);
-            if (same_cache)
-                pl_cache_load_file(p->icc_cache, cache);
-            fclose(cache);
-            if (ret < 0)
-                MP_WARN(p, "Failed loading cache from %s\n", shader_cache);
-        }
-        talloc_free(shader_cache);
-    }
-    if (icc_cache && !same_cache) {
-        FILE *cache = fopen(icc_cache, "rb");
-        if (cache) {
-            int ret = pl_cache_load_file(p->icc_cache, cache);
-            fclose(cache);
-            if (ret < 0)
-                MP_WARN(p, "Failed loading cache from %s\n", icc_cache);
-        }
+    char filename[17];
+    int ret = sprintf(filename, "%016" PRIx64, key);
```

Comment:
sprintf is kind of evil

Reply:
constant width output, static buffer, it's fine

Reply:
Sure it might be fine right now, but what if something changes (intentionally or not) to make it not fine? It's just asking for trouble. One extra argument for `snprintf` can only help.

Reply:
changed